Code/Framework/AzFramework/Platform/Windows/AzFramework/Device/DeviceAttributesCommon_Win.cpp

```diff
@@ -10,37 +10,130 @@
 #include <AzCore/std/string/conversions.h>
 #include <AzFramework/Device/DeviceAttributeDeviceModel.h>
 #include <AzFramework/Device/DeviceAttributeRAM.h>
-#include <AzCore/Console/ILogger.h>
+
+#include <Wbemidl.h>
+#pragma comment(lib, "wbemuuid.lib")
 
 namespace AzFramework
 {
-    DeviceAttributeDeviceModel::DeviceAttributeDeviceModel()
+    AZStd::string QueryWMIProperty(IWbemServices* services, const AZStd::wstring& propertyName)
     {
-        HKEY hKey = nullptr;
-        LSTATUS returnCode = ::RegOpenKeyEx(HKEY_LOCAL_MACHINE, TEXT(R"(HARDWARE\DESCRIPTION\System\BIOS)"), 0, KEY_READ, &hKey);
-        if (returnCode == ERROR_SUCCESS)
+        AZStd::string result;
+        IEnumWbemClassObject* classObjectEnumerator = nullptr;
+
+        // prepare a query to obtain the Model property from Win32_ComputerSystem
+        // https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/win32-computersystem
+        auto propertyQuery = AZStd::wstring::format(TEXT("SELECT %.*s FROM Win32_ComputerSystem"), AZ_STRING_ARG(propertyName));

```

Comment:
If the propertyName is an `AZStd::string_view` parameter it can be formatted using the [AZ_TRAIT_FORMAT_STRING_WPRINTF_STRING](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Platform/Windows/AzCore/AzCore_Traits_Windows.h#L117) macro
```suggestion
        auto propertyQuery = AZStd::wstring::format(TEXT("SELECT " AZ_TRAIT_FORMAT_STRING_WPRINTF_STRING " FROM Win32_ComputerSystem"), AZ_STRING_ARG(propertyName));

```

Reply:
woah thats a new one for me, sure!